### PR TITLE
Use auth lambda 180

### LINKED ISSUE #180: The UI should hit the Gateway API and a Lambda function when a user logs in to Github
Replaces what the Express server currently does with Github Application Key/Secret

### PR DESCRIPTION
**Pull request by Hopding**
Fixes #180. This PR adds an AWS lambda that will take a GitHub authorization token and return an access token for the app. It also removes the `/auth` endpoint from the express backend, and points to the API instead. Note that you will need to add the `REACT_APP_API_URL` env var to your `.env` file to login with this branch.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -18,3 +18,6 @@ yarn-error.log*
 
 # redis
 /data/redis
+
+# AWS Lambda
+/awslambda/github-access-code-getter/node_modules
```
Comment (dane-johnson) [inline .gitignore]: You would do well to generalize this to all directories named node_modules.

File: awslambda/github-access-code-getter/index.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+'use strict'
```
Comment (dane-johnson) [inline awslambda/github-access-code-getter/index.js]: The function name breaks the camelcase naming convention

File: server/server.js [hunk 3]
```diff
@@ -9,19 +9,6 @@ var axios = require('axios')
 var _ = require('lodash');
 _.mixin(require('congruence'));
```
Comment (dane-johnson) [inline server/server.js]: Remove unnecessary imports, and remove the dependencies that are no longer needed.

**Comment by dane-johnson** [comment]
So were still using Heroku for storing of snippets?

File: circle.yml [hunk 4]
```diff
@@ -9,8 +9,14 @@ deployment:
   release:
     tag: /v[0-9]+(\.[0-9]+)*/
     commands:
-        - ./publish release $CIRCLE_TAG
+        - yarn build:
+            environment:
+              REACT_APP_API_URL: api.codesplain.io/prod
```
Comment (thebho) [inline circle.yml]: I don't think we want the prod API to be a `/prod` path.  I would say `/api` or i can look into having no path route to the prod API

File: circle.yml [hunk 5]
```diff
@@ -9,8 +9,14 @@ deployment:
...
+        - ./publish codesplain-ui
```
Comment (thebho) [inline circle.yml]: This is a regressio

Comment (Hopding) [inline circle.yml]: How so? What should it be?

Comment (thebho) [inline circle.yml]: ` - ./publish release $CIRCLE_TAG`  It's pretty obvious in the diff

**Pull request merged**